client/luigi-client.js

```diff
@@ -267,6 +278,42 @@
           return this;
         },
 
+        /** @lends linkManager */
+        /**
+         * Checks if a path exists in the main application, i.e., if that path can be navigated to. This helper method can be used e.g. to conditionally display a DOM element like a button.
+         * @param {string} path path which existence you want to check
+         * @returns {promise} A promise that once checked if a path exists will execute success handler if the path exists, or the error handler otherwise.
+         * @example
+         *  let pathExists;
+         *  this.luigiClient
+         *  .linkManager()
+         *  .pathExists('projects/pr2')
+         *  .then(
+         *    () => { pathExists = true; },
+         *    () => { pathExists = false; },
+         *  );
+         */
+        pathExists: function pathExists(path) {
+          var currentId = Date.now();
+          pathExistsPromises[currentId] = {
+            success: function() {},
+            error: function() {},
+            then: function(success, error) {
```

Comment:
Is there a reason that the user can provide success and error functions instead of just having one simple `pathExists().then((result: Boolean) => {})`. I know it represents a Promise interface with "resolve, reject", but it feels a bit complicated.

Reply:
I initially thought of using it directly inside a template, but since it is out of Zone.js it would not work anyway. So it is fine for me.